### PR TITLE
feat(SPV-989): hide "record transaction" button for admin

### PR DESCRIPTION
**Pull request by swierzbicki**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: src/routes/admin/_admin.transactions.tsx [hunk 1]
```diff
@@ -82,7 +84,7 @@ export function Transactions() {
             <TabsTrigger value="all">All</TabsTrigger>
           </TabsList>
           <div className="flex">
-            <RecordTxDialogAdmin />
+            {showRecordTransaction && <RecordTxDialogAdmin />}
```
Comment (Nazarii-4chain) [inline src/routes/admin/_admin.transactions.tsx:87]: if you need to hide `Record transaction` button, you can remove this component completely from `_admin.transactions.tsx`

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx [hunk 2]
```diff
@@ -31,8 +33,12 @@ export const TransactionsTabContent = ({ transactions, TxDialog, EditDialog }: T
         ) : (
           <div className="flex flex-col items-center gap-1 text-center">
             <h3 className="text-2xl font-bold tracking-tight">You have no Transactions</h3>
-            <p className="text-sm text-muted-foreground mb-2">You can record Transaction here.</p>
-            <TxDialog />
+            {showRecordTransaction && (
+              <>
+                <p className="text-sm text-muted-foreground mb-2">You can record Transaction here.</p>
+                <TxDialog />
```
Comment (Nazarii-4chain) [inline src/components/TransactionsTabContent/TransactionsTabContent.tsx:39]: Make `<TxDialog />` optional in `TransactionsTabContentProps` interface. Then check if `TxDialog` is defined.

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx [hunk 3]
```diff
@@ -31,8 +33,12 @@ export const TransactionsTabContent = ({ transactions, TxDialog, EditDialog }: T
         ) : (
           <div className="flex flex-col items-center gap-1 text-center">
             <h3 className="text-2xl font-bold tracking-tight">You have no Transactions</h3>
-            <p className="text-sm text-muted-foreground mb-2">You can record Transaction here.</p>
-            <TxDialog />
+            {showRecordTransaction && (
```
Comment (Nazarii-4chain) [inline src/components/TransactionsTabContent/TransactionsTabContent.tsx:36]: Change this to `TxDialog`

File: src/routes/admin/_admin.transactions.tsx [hunk 4]
```diff
@@ -62,6 +62,8 @@ export function Transactions() {
   const [debouncedBlockHeight] = useDebounce(blockHeight, 200);
   const { order_by_field, sort_direction } = useSearch({ from: '/admin/_admin/transactions' });
 
+  const [showRecordTransaction] = useState(false);
```
Comment (Nazarii-4chain) [inline src/routes/admin/_admin.transactions.tsx:65]: You don't need state here.

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx [hunk 5]
```diff
@@ -20,6 +20,8 @@ export interface TransactionsTabContentProps {
 }
 
 export const TransactionsTabContent = ({ transactions, TxDialog, EditDialog }: TransactionsTabContentProps) => {
+  const [showRecordTransaction] = useState(false);
```
Comment (Nazarii-4chain) [inline src/components/TransactionsTabContent/TransactionsTabContent.tsx:23]: Also you can remove state here.

**Review by Nazarii-4chain — CHANGES_REQUESTED** [review]

File: src/routes/admin/_admin.transactions.tsx (hunk 4 above)
Comment (swierzbicki) [reply src/routes/admin/_admin.transactions.tsx:65]: Sure. I thought that maybe we will want to `setShowRecordTransaction` in the future. But agree. I will leave just a simple boolean flag.

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx (hunk 5 above)
Comment (swierzbicki) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:23]: Same here

>Sure. I thought that maybe we will want to `setShowRecordTransaction` in the future. But agree. I will leave just a simple boolean flag.

File: src/routes/admin/_admin.transactions.tsx (hunk 1 above)
Comment (dorzepowski) [reply src/routes/admin/_admin.transactions.tsx:87]: I don't agree, 
if we will remove this component, after some time, no one would remember that it is already there, and then we can actually remove the entire feature from code.

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx (hunk 2 above)
Comment (dorzepowski) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:39]: Why?

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx (hunk 3 above)
Comment (dorzepowski) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:36]: Why?

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx [hunk 6]
```diff
@@ -20,6 +20,8 @@ export interface TransactionsTabContentProps {
 }
 
 export const TransactionsTabContent = ({ transactions, TxDialog, EditDialog }: TransactionsTabContentProps) => {
+  const showRecordTransaction = false;
```
Comment (dorzepowski) [inline src/components/TransactionsTabContent/TransactionsTabContent.tsx:23]: Please add jsdoc, that we're hiding record transaction button and dialog, until spv-wallet functionality for recording transactions would fulfil users needs and expectations.

File: src/routes/admin/_admin.transactions.tsx [hunk 7]
```diff
@@ -62,6 +62,8 @@ export function Transactions() {
   const [debouncedBlockHeight] = useDebounce(blockHeight, 200);
   const { order_by_field, sort_direction } = useSearch({ from: '/admin/_admin/transactions' });
 
+  const showRecordTransaction = false;
```
Comment (dorzepowski) [inline src/routes/admin/_admin.transactions.tsx:65]: Please add jsdoc, that we're hiding record transaction button and dialog, until spv-wallet functionality for recording transactions would fulfil users needs and expectations.

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx (hunk 2 above)
Comment (Nazarii-4chain) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:39]: Passing TxDialog as a prop will define if TxDialog should be rendered.

File: src/routes/admin/_admin.transactions.tsx (hunk 7 above)
Comment (swierzbicki) [reply src/routes/admin/_admin.transactions.tsx:65]: Sure.

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx (hunk 6 above)
Comment (swierzbicki) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:23]: Sure.

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx (hunk 2 above)
Comment (dorzepowski) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:39]: And how would it make it clearer for reader or easier to maintain when comparing to flag with a documentation?

File: src/routes/admin/_admin.transactions.tsx (hunk 1 above)
Comment (Nazarii-4chain) [reply src/routes/admin/_admin.transactions.tsx:87]: Ok.
PS: I like the idea with `display: none` or `visibility: hidden` more :blush:

Comment (swierzbicki) [reply src/routes/admin/_admin.transactions.tsx:92]: And how will you show that this change is on purpose? 

For me `display:none` has the same "value" as commented code. There is no value in terms of explaining why we hide specific feature.

File: src/components/TransactionsTabContent/TransactionsTabContent.tsx (hunk 2 above)
Comment (Nazarii-4chain) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:44]: Passing props it's a common practice in React projects. One of the use cases is a conditional rendering. This is what we have here. If you want to use flag - no problem

Comment (dorzepowski) [reply src/components/TransactionsTabContent/TransactionsTabContent.tsx:44]: I know that it's common practice, but the question was how does it make the intention of the code clearer.
The flag with documentation why it's hidden for me it's clear from the first look. While, if I would look for the first time, and see some optional parameter, which isn't passed at all, would either think that we should remove this optional parameter or I would ask a question to my self WTF.

**Review by dorzepowski — APPROVED** [review]

**Review by Nazarii-4chain — APPROVED** [review]

**Pull request merged**